frontend/src/views/coverage/enrollment/ResidentDetails.vue

```diff
@@ -0,0 +1,385 @@
+import { API_DATE_FORMAT } from '../../../util/constants'
+import { formatPersonName } from '../../../util/utils'
+
+export default {
+  name: 'ResidentDetails',
+  props: {
+    resident: {
+      phn: '',
+      givenName: '',
+      secondName: '',
+      surname: '',
+      dateOfBirth: '',
+    },
+  },
+  components: {
+    AppButton,
+    AppCol,
+    AppDateInput,
+    AppInput,
+    AppOutput,
+    AppRow,
+    AppSelect,
+  },
+  setup() {
+    return {
+      v$: useVuelidate(),
+    }
+  },
+  data() {
+    return {
+      searching: false,
+      searchOk: false,
+      //Add Visa Resident Fields
+      groupNumber: '',
+      immigrationCode: '',
+      groupMemberNumber: '',
+      visaIssueDate: null,
+      departmentNumber: '',
+      visaExpiryDate: null,
+      residenceDate: null,
+      coverageEffectiveDate: null,
+      telephone: '',
+      coverageCancellationDate: null,
+      address1: '',
+      address2: '',
+      address3: '',
+      city: '',
+      province: '',
+      postalCode: '',
+      mailingAddress1: '',
+      mailingAddress2: '',
+      mailingAddress3: '',
+      mailingAddressCity: '',
+      mailingAddressProvince: '',
+      mailingAddressPostalCode: '',
+      priorResidenceCode: '',
+      otherProvinceHealthcareNumber: '',
+      // Immigration Code drop down options
+      immigrationCodeOptions: [
+        { text: '', value: '' },
+        { text: 'Student Authorization', value: 'S' },
+      ],
+      // Province drop down options
+      provinceOptions: [
+        { text: '', value: '' },
+        { text: 'Alberta', value: 'AB' },
+        { text: 'British Columbia', value: 'BC' },
+        { text: 'Manitoba', value: 'MB' },
+        { text: 'New Brunswick', value: 'NB' },
+        { text: 'Newfoundland', value: 'NL' },
+        { text: 'Nova Scotia', value: 'NS' },
+        { text: 'Northwest Territories', value: 'NT' },
+        { text: 'Nunavut', value: 'NU' },
+        { text: 'Ontario', value: 'ON' },
+        { text: 'P.E.I', value: 'PE' },
+        { text: 'Quebec', value: 'QC' },
+        { text: 'Saskatchewan', value: 'SK' },
+        { text: 'Yukon', value: 'YT' },
+      ],
+      // Prior Residence drop down options
+      priorResidenceOptions: [
+        { text: '', value: '' },
+        { text: 'Alberta', value: 'AB' },
+        { text: 'Manitoba', value: 'MB' },
+        { text: 'New Brunswick', value: 'NB' },
+        { text: 'Newfoundland', value: 'NL' },
+        { text: 'Nova Scotia', value: 'NS' },
+        { text: 'Northwest Territories', value: 'NT' },
+        { text: 'Nunavut', value: 'NU' },
+        { text: 'Other Country', value: 'OC' },
+        { text: 'Ontario', value: 'ON' },
+        { text: 'P.E.I', value: 'PE' },
+        { text: 'Quebec', value: 'QC' },
+        { text: 'Saskatchewan', value: 'SK' },
+        { text: 'U.S.A', value: 'US' },
+        { text: 'Yukon', value: 'YT' },
+        { text: 'British Columbia', value: 'BC' },
+      ],
```

Comment:
Will this be used in other screens? If so maybe refactor? Also it's technically not reactive so doesn't need to be in data.

Reply:
This is very custom to this screen. Moved out of data to computed.